rest_sensors.cpp

```diff
@@ -2171,28 +2171,45 @@ int DeRestPluginPrivate::changeSensorConfig(const ApiRequest &req, ApiResponse &
-                            else if (ok && mode == 3) { interfaceMode = PULSE_COUNTING_WATER; }
-                            else if (ok && mode == 4) { interfaceMode = KAMSTRUP_KMP; }
-                            else if (ok && mode == 5) { interfaceMode = LINKY; }
-                            else if (ok && mode == 6) { interfaceMode = DLMS_COSEM; }
-                            else if (ok && mode == 7) { interfaceMode = DSMR_23; }
-                            else if (ok && mode == 8) { interfaceMode = DSMR_40; }
-                            else
+                            if(sensor->modelId() == QLatin1String("ZHEMI101"))
+                            {
+                                availableModes = 8;
+                                
+                                if      (ok && mode == 1) { interfaceMode = PULSE_COUNTING_ELECTRICITY; }
+                                else if (ok && mode == 2) { interfaceMode = PULSE_COUNTING_GAS; }
+                                else if (ok && mode == 3) { interfaceMode = PULSE_COUNTING_WATER; }
+                                else if (ok && mode == 4) { interfaceMode = KAMSTRUP_KMP; }
+                                else if (ok && mode == 5) { interfaceMode = LINKY; }
+                                else if (ok && mode == 6) { interfaceMode = DLMS_COSEM; }
+                                else if (ok && mode == 7) { interfaceMode = DSMR_23; }
+                                else if (ok && mode == 8) { interfaceMode = DSMR_40; }
+                            }
+                            else if (sensor->modelId().startsWith(QLatin1String("EMIZB-1")))
+                            {
+                                availableModes = 5;
+                                
+                                if      (ok && mode == 1) { interfaceMode = NORWEGIAN_HAN; }
+                                else if (ok && mode == 2) { interfaceMode = NORWEGIAN_HAN_EXTRA_LOAD; }
+                                else if (ok && mode == 3) { interfaceMode = AIDON_METER; }
+                                else if (ok && mode == 4) { interfaceMode = KAIFA_KAMSTRUP_METERS; }
+                                else if (ok && mode == 5) { interfaceMode = AUTO_DETECT; }
+                            }
+                            
+                            if(interfaceMode == 0)
                             {
                                 rsp.list.append(errorToMap(ERR_INVALID_VALUE, QString("/sensors/%1/config/%2").arg(id).arg(pi.key()),
                                                            QString("invalid value, %1, for parameter %2").arg(map[pi.key()].toString()).arg(pi.key())));
                                 rsp.httpStatus = HttpStatusBadRequest;
                                 return REQ_READY_SEND;
                             }
 
-                            if (mode > 0 && mode < 9)
+                            if (availableModes && mode > 0 && mode <= availableModes)
```

Comment:
If I understand the whole code section correctly..

Why not ditch the `availableModes` completely to simplify the code? 

We can only reach at this point when `interfaceMode != 0` due the error checking above, so even an invalid mode from the client is already handled. The `if (availableModes && mode > 0 && mode <= availableModes)` check isn't needed, it's like a duplicated check for the same thing?

Reply:
Done